service/controller/managementcluster/controller.go

```diff
@@ -84,17 +86,26 @@ func NewController(config ControllerConfig) (*Controller, error) {
 
 	var operatorkitController *controller.Controller
 	{
+		selectors := labels.Set{
+			"component": "apiserver",
+		}
+		if key.IsCAPIManagementCluster(config.Provider) {
+			selectors = labels.Set{
+				"cluster.x-k8s.io/cluster-name": config.Installation,
+			}
+		}
 		c := controller.Config{
 			K8sClient: config.K8sClient,
 			Logger:    config.Logger,
 			Name:      project.Name() + "-management-cluster-controller",
 			NewRuntimeObjectFunc: func() client.Object {
+				if key.IsCAPIManagementCluster(config.Provider) {
```

Comment:
If the Management cluster is CAPI only, we reconcile the cluster CR of the MC, otherwise we reconcile the kubernetes service